service/s3control/api_client.go

```diff
@@ -276,6 +276,22 @@ func addResponseErrorMiddleware(stack *middleware.Stack) error {
 	return s3shared.AddResponseErrorMiddleware(stack)
 }
 
+// ResponseError provides the HTTP centric error type wrapping the underlying error
+// with the HTTP response value and the deserialized RequestID.
+type ResponseError interface {
+	error
+
+	ServiceHostID() string
+	ServiceRequestID() string
+}
+
+// GetHostIDMetadata retrieves the host id from middleware metadata returns host id
+// as string along with a boolean indicating presence of hostId on middleware
+// metadata.
+func GetHostIDMetadata(metadata middleware.Metadata) (string, bool) {
+	return s3shared.GetHostIDMetadata(metadata)
+}
```

Comment:
In case of an error, do we not return this `middleware.Metadata`? If not, should we?

Reply:
The metadata is primarily for the response since the errors can use layering and dont use the metadata. Though we could consider expanding the `smithy.OperationFailure` error to include metadata

Reply:
https://github.com/awslabs/smithy-go/issues/243